redwood-protocol-widget/src/commonMain/kotlin/app/cash/redwood/protocol/widget/ProtocolBridge.kt

```diff
@@ -75,37 +75,26 @@ public class ProtocolBridge<W : Any>(
     for (layoutModifier in diff.layoutModifiers) {
       val node = node(layoutModifier.id)
       val childIndex = node(node.parentId).childIds.indexOf(layoutModifier.id)
-      node.widget.updateLayoutModifier(layoutModifier.elements)
+      node.updateLayoutModifier(layoutModifier.elements)
       node.parentChildren.onLayoutModifierUpdated(childIndex)
```

Comment:
This logic can be dramatically cleaned up if we stop delivering `LayoutModifiers` to the `Widget` and _only_ deliver them to the parent. The protocol entry for layout modifier changes can reference the parent ID, the children tag, and the index inside that set of children on which the new value applies. That will eliminate nearly 100% of the `parentId`/`parentChildren` wiring present in this change.